viostor/virtio_pci.c

```diff
@@ -197,12 +197,12 @@ static u16 vdev_get_msix_vector(void *context, int queue)
             } else {
                 vector = queue + 1;
             }
-        } else {
-            vector = VIRTIO_MSI_NO_VECTOR;
         }
     } else {
         /* on-device-config-change interrupt */
-        vector = VIRTIO_MSI_NO_VECTOR;
+        if (!adaptExt->msix_one_vector) {
```

Comment:
If the device has only one vector and the driver requires more, then the driver should fail to load.

The current state of viostor is that it loads, pretends to support one vector (the msix_one_vector flag which you added recently), but doesn't function properly. That's certainly a bug.

The fix is either to have the driver fail to load if it doesn't have enough vectors, or fix the code to be able to drive a one-vector device. I don't see why it wouldn't be possible - you just have to look at all sources of interrupts when you receive the one message.